### PR TITLE
[Publisher][Admin Panel] Auto-add users/agencies created via second flow

### LINKED ISSUE #1146: [Publisher] Admin Panel: Auto-add newly created users/agencies when they are created via the second modal flow
When you're in a primary modal to create/edit a user/agency, and decide to create either a user/agency via the second modal flow, it would be nice if the user/agency you just created is auto-added to the primary modal because that's the primary intention of creating an entity via the second modal. This will help save an extra step of manually adding the user/agency after creating them.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

1. Auto-add logic for users/agencies created via second flow by id

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

closes #1146 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by nasaownsky** [comment]
Hi @mxosman! So I've created the logic to auto add new users/agencies created via second flow to primary flow, but I've been struggling with getting actual `id` of newly created user/agency. As I understand `saveAgencyProvisioningUpdates` is capable to return json response, but in actual provisioning components it returns fetch type because it's need it there. Should I try to create new getter in `AdminPanelStore`?

**Comment by mxosman** [comment]
> Hi @mxosman! So I've created the logic to auto add new users/agencies created via second flow to primary flow, but I've been struggling with getting actual `id` of newly created user/agency. As I understand `saveAgencyProvisioningUpdates` is capable to return json response, but in actual provisioning components it returns fetch type because it's need it there. Should I try to create new getter in `AdminPanelStore`?

Hi @nasaownsky! Hmm, dang - that's a good point. I'll need to think about this a bit more, but please feel free to try any solution you think would work best! I have no issues with adding a getter/updating the store to accommodate. 

If it feels like it would require a much bigger refactor though, let me know and we can think of other solutions.

File: publisher/src/stores/AdminPanelStore.ts [hunk 1]
```diff
@@ -151,6 +153,10 @@ class AdminPanelStore {
       });
   }
 
+  get createdUserResponse(): UserResponse | undefined {
+    return this.userResponse;
+  }
+
```
Comment (nasaownsky) [inline publisher/src/stores/AdminPanelStore.ts:159]: @mxosman it seems like this approach is working, check my changes and please leave your thoughts about implementation, namings, etc. Thank you!

Comment (mxosman) [reply publisher/src/stores/AdminPanelStore.ts:159]: Oo - I'm excited to check this out - probably later today or tomorrow morning at the latest! Thank you, Ilya!

File: publisher/src/components/AdminPanel/AgencyProvisioning.tsx [hunk 2]
```diff
@@ -302,8 +304,14 @@ export const AgencyProvisioning: React.FC<ProvisioningProps> = observer(
           show: false,
           errorMessage: undefined,
         }));
-        if (response && "status" in response && response.status === 200)
+        if (response && "status" in response && response.status === 200) {
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/AgencyProvisioning.tsx:307]: Ugh... sorry, I'm sure this was a bit ugly to work within.

File: publisher/src/components/AdminPanel/AgencyProvisioning.tsx [hunk 3]
```diff
@@ -302,8 +304,14 @@ export const AgencyProvisioning: React.FC<ProvisioningProps> = observer(
...
+          const agencyResponse = adminPanelStore.createdAgencyResponse;
+          if (setSecondaryCreatedId && agencyResponse) {
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/AgencyProvisioning.tsx:309]: Brilliant! Maybe we can add a brief comment here and in the UserProvisioning component explaining that if this view is the secondary create modal, then we'll store the newly created ID for the purpose of auto-adding after creation - or something like that.

File: publisher/src/components/AdminPanel/UserProvisioning.tsx [hunk 4]
```diff
@@ -105,6 +107,13 @@ export const UserProvisioning: React.FC<ProvisioningProps> = observer(
         : []),
     ];
 
+    useEffect(() => {
+      if (secondaryCreatedId)
+        setAddedAgenciesIDs((prev) =>
+          toggleAddRemoveSetItem(prev, +secondaryCreatedId)
+        );
+    }, [secondaryCreatedId]);
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/UserProvisioning.tsx:115]: Love this! I would just add a brief comment that this is where the auto-adding happens if something was created via the secondary modal.

File: publisher/src/components/AdminPanel/UserProvisioning.tsx [hunk 5]
```diff
@@ -227,8 +236,14 @@ export const UserProvisioning: React.FC<ProvisioningProps> = observer(
       /** After showing the confirmation screen, either return to modal (on error) or close modal (on success) */
       setTimeout(() => {
         setShowSaveConfirmation((prev) => ({ ...prev, show: false }));
-        if (response && "status" in response && response.status === 200)
+        if (response && "status" in response && response.status === 200) {
+          const userResponse = adminPanelStore.createdUserResponse;
+          if (setSecondaryCreatedId && userResponse) {
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/UserProvisioning.tsx:241]: Same as a earlier comment re: adding a small comment here.

**Review by mxosman — APPROVED** [review]
Wow @nasaownsky! What a fantastic solution - I never would've thought to do it this way. It feels great when I was playing around with it - and works well when I create multiple agencies and select existing ones within the same modal session. [Sorry I didn't ever get around to refactoring the Admin Panel components - I hope it wasn't too much of a headache to work within!]

After adding a few comments - please feel free to merge after!

**Comment by mxosman** [comment]
Oh one last thing @nasaownsky - can you also include @lilidworkin on future PRs as a reviewer? Many thanks in advance!

**Comment by nasaownsky** [comment]
Thank you for great suggestions @mxosman! Comments added. Also I will definitely including Lili as a reviewer in the future, don't know why I didn't do it here.

**Pull request merged**